server/service/testing_utils.go

```diff
@@ -174,6 +179,15 @@ func newTestServiceWithConfig(t *testing.T, ds fleet.Datastore, fleetConfig conf
 		panic(err)
 	}
 	if lic.IsPremium() {
+		if softwareInstallStore == nil {
+			// default to file-based
+			dir := t.TempDir()
+			store, err := filesystem.NewSoftwareInstallerStore(dir)
+			if err != nil {
+				panic(err)
+			}
+			softwareInstallStore = store
```

Comment:
When we add the upload/download API for software installers, we should add integration tests that use an S3-based (minio) storage so that the recommended store is well-tested in end-to-end scenarios.

Reply:
for our future reference, this is how the sandbox integration tests do it:

https://github.com/fleetdm/fleet/blob/c936e453a03506aa31ec360b2f1aba8ae481f903/server/service/integration_sandbox_test.go#L36-L37

Reply:
Thanks! I added the `TestServerOpts` support in this PR so it will be ready to receive a proper `SoftwareInstallerStore` in the integration test.